src/util/tumblr_helpers.js

```diff
@@ -7,7 +7,21 @@ import { inject } from './inject.js';
  */
 export const apiFetch = async function (...args) {
   return inject(
-    async (resource, init) => window.tumblr.apiFetch(resource, init),
+    async (resource, init) => {
+      // make sure init is an object we can modify
+      if (typeof init !== 'object' || init === null) {
+        init = {};
+      }
+
+      // make sure init.headers is an object we can modify
+      if (typeof init.headers !== 'object' || init.headers === null) {
+        init.headers = {};
+      }
```

Comment:
i'd say it's acceptable given the lacking jsdoc. (hopefully the eslint plugin doesn't complain about me documenting specific arguments within rest parameters...)

that said, it would definitely be possible to make use of some newer and cuter language features here, since XKit Rewritten has very strict browser requirements:
1. it's safe to assume `init` is either going to be an object or omitted entirely, so ensuring it exists is as simple as setting it as a [default parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)
2. it's also fairly safe to assume `init.headers` is either going to be an object or omitted (I'm sure setting it to another type would likely have some adverse effects), so ensuring it exists is also straightforward using [logical nullish assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment)

```suggestion
    async (resource, init = {}) => {
      init.headers ??= {}; 
```

Reply:
> [logical nullish assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment)

Ooh, I was avoiding using that one because I wasn't sure if people would be familiar with it! Will make use in future :)

Reply:
awesome, thank you for explaining that so thoroughly 👌